vault/core_metrics.go

```diff
@@ -514,3 +517,13 @@ func (c *Core) cachedGaugeMetricsEmitter() {
 
 	loopMetrics.Range(emit)
 }
+
+func (c *Core) inFlightReqGaugeMetric() {
+	totalInFlightReq := 0
+	c.inFlightReqMap.Range(func(key, value interface{}) bool {
```

Comment:
I wonder if it's worth having a separate atomic int to track the request count, rather than having to range across the map every second?